pkg/controller/pxc/controller.go

```diff
@@ -1328,6 +1328,38 @@ func (r *ReconcilePerconaXtraDBCluster) createOrUpdate(obj client.Object) error
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) create(obj client.Object) error {
+	metaAccessor, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		return errors.New("can't convert object to ObjectMetaAccessor")
+	}
+
+	objectMeta := metaAccessor.GetObjectMeta()
+	if objectMeta.GetAnnotations() == nil {
+		objectMeta.SetAnnotations(make(map[string]string))
+	}
+
+	objAnnotations := objectMeta.GetAnnotations()
+	delete(objAnnotations, "percona.com/last-config-hash")
+	objectMeta.SetAnnotations(objAnnotations)
+
+	hash, err := getObjectHash(obj)
+	if err != nil {
+		return errors.Wrap(err, "calculate object hash")
+	}
+
+	objAnnotations = objectMeta.GetAnnotations()
+	objAnnotations["percona.com/last-config-hash"] = hash
+	objectMeta.SetAnnotations(objAnnotations)
+
+	err = r.client.Create(context.TODO(), obj)
```

Comment:
This is just my curiosity, not related to these changes, but I was wondering why we pass TODO contexts all over the place, why don't we create a proper context and pass it down?